sdks/python/apache_beam/transforms/trigger.py

```diff
@@ -972,10 +988,13 @@ def __eq__(self, other):
     if isinstance(other, collections.Iterable):
       return all(
           a == b
-          for a, b in itertools.izip_longest(self, other, fillvalue=object()))
+          for a, b in zip_longest(self, other, fillvalue=object()))
     else:
       return NotImplemented
 
+  def __hash__(self):
+    return hash(self)
```

Comment:
This will recurse infinitely, I think.

Reply:
Exactly thx! Fixed now with: `return hash(tuple(self))`